neptune/new/internal/backends/operations_preprocessor.py

```diff
@@ -52,17 +52,26 @@
 class OperationsPreprocessor:
     def __init__(self):
         self._accumulators = dict()
+        self.processed_ops_count = 0
 
     def process(self, operations: List[Operation]):
         for op in operations:
-            path_str = path_to_str(op.path)
-            self._accumulators.setdefault(
-                path_str, _OperationsAccumulator(op.path)
-            ).visit(op)
-        result = []
-        for acc in self._accumulators.values():
-            result.extend(acc.get_operations())
-        return result
+            acc: "_OperationsAccumulator" = self._process_op(op)
```

Comment:
Instead of maintaining `has_file_ops` it would be easier to just stop processing operations when file operation occur and there already exists any delete operation, wouldn't it?
Let's just throw exception in such case and handle it here instead of doing some reversing of already processed operations. What do you think?

Reply:
I guess my initial idea was to encapsulate the decision process to the `_OperationsAccumulator` -- but it's leaking either way, so what you're proposing would probably result in a cleaner code. I'll try and restructure.